TCAT/Cells/LargeDetailTableViewCell.swift

```diff
@@ -15,214 +15,188 @@ protocol LargeDetailTableViewDelegate: class {
+        // Remove the container as it is no longer needed
+        containerView.removeFromSuperview()
 
-        return label
+        return iconAttachment
     }
 
-    /** Abstracted formatting of content for detailLabel. Needs titleLabel */
-    func formatDetailLabel(_ label: UILabel, _ titleLabel: UILabel) -> UILabel {
+    private func formatTitleLabel(for direction: Direction) {
+        let paragraphStyle = NSMutableParagraphStyle()
 
-        label.text = "\(direction.stops.count + 1) stop\(direction.stops.count + 1 == 1 ? "" : "s")"
+        // Beginning of string
+        let titleLabelText = NSMutableAttributedString(string: direction.type == .transfer ? "Bus becomes" : "Board")
 
-        // Number of minutes for the bus direction
-        var timeString = Time.timeString(from: direction.startTime, to: direction.endTime)
-        if timeString == "0 min" {  timeString = "1 min" }
-        label.text = label.text! +  " • \(timeString)"
-
-        label.sizeToFit()
-        label.frame.origin.y = titleLabel.frame.maxY + labelSpacing
-        return label
-    }
-
-    /** Abstracted formatting of content for busIconView. Needs initialized titleLabel */
-    func formatBusIconView(_ icon: BusIcon, _ titleLabel: UILabel) -> BusIcon {
+        titleLabelText.append(NSAttributedString(attachment: getBusIconImageAsTextAttachment(for: direction)))
 
-        let plainLabel = getTitleLabel()
-        let originX = titleLabel.frame.minX + plainLabel.frame.size.width + 8
-        var originY = titleLabel.frame.minY
+        // Append rest of string
+        let content = direction.locationNameDescription
+        let labelBoldFont: UIFont = .getFont(.semibold, size: 14)
+        let attributedString = direction.name.bold(in: content, from: titleLabel.font, to: labelBoldFont)
+        titleLabelText.append(attributedString)
+        titleLabel.attributedText = titleLabelText
 
-        originY += titleLabel.font.lineHeight - icon.frame.size.height / 2 - CGFloat(titleLabel.numberOfLines() * 2)
+        paragraphStyle.lineSpacing = 4
 
-        icon.frame.origin = CGPoint(x: originX, y: originY)
-        return icon
+        titleLabel.numberOfLines = 0
+        titleLabelText.addAttribute(.paragraphStyle,
+                                    value: paragraphStyle,
+                                    range: NSRange(location: 0, length: titleLabel.attributedText!.length))
+        titleLabel.attributedText = titleLabelText
     }
 
-    /** Precondition: setCell must be called before using this function */
-    func height() -> CGFloat {
-        let titleLabel = formatTitleLabel(getTitleLabel())
-        let detailLabel = formatDetailLabel(getDetailLabel(), titleLabel)
-        return titleLabel.frame.height + detailLabel.frame.height + labelSpacing + (edgeSpacing * 2)
+    private func formatDetailLabel(for direction: Direction) {
+        let detailLabelText = "\(direction.stops.count + 1) stop\(direction.stops.count + 1 == 1 ? "" : "s")"
```

Comment:
can we store `direction.stops.count + 1` in some variable since we use it twice?